pennylane/estimator/estimate.py

```diff
@@ -361,9 +362,23 @@ def _get_decomposition(
         A tuple containing the decomposition function and its associated kwargs.
     """
     op_type = comp_res_op.op_type
-    # TODO: Restore logic to handle symbolic operators when symboli operators are merged.
 
-    kwargs = config.resource_op_precisions.get(op_type, {})
-    decomp_func = config.custom_decomps.get(op_type, op_type.resource_decomp)
+    _SYMBOLIC_DECOMP_MAP = {
+        Adjoint: "_adj_custom_decomps",
+        Controlled: "_ctrl_custom_decomps",
+        Pow: "_pow_custom_decomps",
+    }
+
+    if op_type in _SYMBOLIC_DECOMP_MAP:
+        decomp_attr_name = _SYMBOLIC_DECOMP_MAP[op_type]
+        custom_decomp_dict = getattr(config, decomp_attr_name)
+
+        base_op_type = comp_res_op.params["base_cmpr_op"].op_type
+        kwargs = config.resource_op_precisions.get(base_op_type, {})
+        decomp_func = custom_decomp_dict.get(base_op_type, op_type.resource_decomp)
+
+    else:
+        kwargs = config.resource_op_precisions.get(op_type, {})
+        decomp_func = config.custom_decomps.get(op_type, op_type.resource_decomp)
```

Comment:
Can we refactor this to use less duplicated code (I know this is my own code but still 😆)
```suggestion
    lookup_op_type = op_type
    custom_decomp_dict = config.custom_decomps
    
    if op_type in _SYMBOLIC_DECOMP_MAP:
        decomp_attr_name = _SYMBOLIC_DECOMP_MAP[op_type]
        custom_decomp_dict = getattr(config, decomp_attr_name)
        lookup_op_type = comp_res_op.params["base_cmpr_op"].op_type
    
    kwargs = config.resource_op_precisions.get(lookup_op_type, {})
    decomp_func = custom_decomp_dict.get(lookup_op_type, op_type.resource_decomp)
```